pkg/component/etcd/etcd/etcd.go

```diff
@@ -228,28 +229,13 @@ func (e *etcd) Deploy(ctx context.Context) error {
 		volumeClaimTemplate = e.values.Role + "-" + strings.TrimSuffix(e.etcd.Name, "-"+e.values.Role)
 	}
 
-	etcdCASecret, found := e.secretsManager.Get(v1beta1constants.SecretNameCAETCD)
-	if !found {
-		return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCAETCD)
-	}
-
-	serverSecret, err := e.secretsManager.Generate(ctx, &secretsutils.CertificateSecretConfig{
-		Name:                        secretNamePrefixServer + e.values.Role,
-		CommonName:                  "etcd-server",
-		DNSNames:                    e.clientServiceDNSNames(),
-		CertType:                    secretsutils.ServerClientCert,
-		SkipPublishingCACertificate: true,
-	}, secretsmanager.SignedByCA(v1beta1constants.SecretNameCAETCD), secretsmanager.Rotate(secretsmanager.InPlace))
-	if err != nil {
-		return err
-	}
-
-	clientSecret, err := e.secretsManager.Generate(ctx, &secretsutils.CertificateSecretConfig{
-		Name:                        SecretNameClient,
-		CommonName:                  "etcd-client",
-		CertType:                    secretsutils.ClientCert,
-		SkipPublishingCACertificate: true,
-	}, secretsmanager.SignedByCA(v1beta1constants.SecretNameCAETCD), secretsmanager.Rotate(secretsmanager.InPlace))
+	etcdCASecret, serverSecret, clientSecret, err := GenerateClientServerCertificates(
+		ctx,
+		e.secretsManager,
+		e.values.Role,
+		e.clientServiceDNSNames(),
+		nil,
+	)
```

Comment:
This could have been a separate commit

Reply:
Separated into multiple commits as requested.